### PR TITLE
Use ClimaCore 0.11 and MPITrampoline

### PR DESCRIPTION
**Pull request by Sbozzolo**

### CODE REVIEW

File: .buildkite/gpu_pipeline/pipeline.yml [hunk 1]
```diff
@@ -4,10 +4,23 @@ agents:
...
+  CONFIG_PATH: "config/model_configs"
   GPU_CONFIG_PATH: "config/gpu_configs/"
+  PERF_CONFIG_PATH: "config/perf_configs"
+  MPI_CONFIG_PATH: "config/mpi_configs"
+  SLURM_KILL_BAD_EXIT: 1
+  JULIA_NVTX_CALLBACKS: gc
+  JULIA_CUDA_MEMORY_POOL: none
+  JULIA_MPI_HAS_CUDA: "true"
+  MPITRAMPOLINE_LIB: "/groups/esm/software/MPIwrapper/ompi4.1.5_cuda-12.2/lib64/libmpiwrapper.so"
+  MPITRAMPOLINE_MPIEXEC: "/groups/esm/software/MPIwrapper/ompi4.1.5_cuda-12.2/bin/mpiwrapperexec"
```
Comment (simonbyrne) [inline .buildkite/gpu_pipeline/pipeline.yml:23]: This path doesn't exist on `clima`

Comment (simonbyrne) [reply .buildkite/gpu_pipeline/pipeline.yml:23]: let me see if i can push a fix.

Comment (Sbozzolo) [reply .buildkite/gpu_pipeline/pipeline.yml:23]: Why is this buildkite pipeline relevant for `clima`?

Comment (simonbyrne) [reply .buildkite/gpu_pipeline/pipeline.yml:23]: the `gpu_pipeline` runs on `clima`, not central.

Comment (Sbozzolo) [reply .buildkite/gpu_pipeline/pipeline.yml:23]: Ah, I didn't realize it!

Good to know!

**Comment by Sbozzolo** [comment]
Closes #2138

**Comment by Sbozzolo** [comment]
I am trying to get MPI to work.

I managed to have it working here: https://buildkite.com/clima/climaatmos-ci/builds/15017

Then, I pushed again the same commit, and it is failing:
https://buildkite.com/clima/climaatmos-ci/builds/15020

The main difference is that in one case the depot was cold, so everything had to be compiled.

**Comment by Sbozzolo** [comment]
In this build:

https://buildkite.com/clima/climaatmos-ci/builds/15043#018bf91f-8c1f-4f82-8a8e-a5b4cd410de8

`CA.solve_atmos!(CA.get_integrator(CA.AtmosConfig())` run on 2 processes, but not the MPI example

**Comment by szy21** [comment]
Maybe change the ` CLIMACORE_DISTRIBUTED: "MPI"` in MPI example to `CLIMACOMMS_CONTEXT: "MPI"`?

**Comment by Sbozzolo** [comment]
> Maybe change the ` CLIMACORE_DISTRIBUTED: "MPI"` in MPI example to `CLIMACOMMS_CONTEXT: "MPI"`?

Failing https://buildkite.com/clima/climaatmos-ci/builds/15045

**Comment by simonbyrne** [comment]
For some reason it is using the wrong preferences:
https://buildkite.com/clima/climaatmos-ci/builds/15080#018bfd3b-6aed-4bf1-8208-d1fecdc8dc1b/160-167

**Comment by Sbozzolo** [comment]
> For some reason it is using the wrong preferences: https://buildkite.com/clima/climaatmos-ci/builds/15080#018bfd3b-6aed-4bf1-8208-d1fecdc8dc1b/160-167

Thanks for the lead, let me investigate this.

**Comment by simonbyrne** [comment]
Ah, I think it's because it's picking up the preferences from the shared depot (which was set previously).
```
├ cat /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/default/environments/v1.9/LocalPreferences.toml
[MPIPreferences]
__clear__ = ["preloads_env_switch"]
_format = "1.0"
abi = "OpenMPI"
binary = "system"
cclibs = []
libmpi = "libmpi"
mpiexec = "mpiexec"
preloads = []
```

either delete that file, or just disable the shared depot for now.

**Comment by Sbozzolo** [comment]
> Ah, I think it's because it's picking up the preferences from the shared depot (which was set previously).
> 
> ```
> ├ cat /central/scratch/esm/slurm-buildkite/climaatmos-ci/depot/default/environments/v1.9/LocalPreferences.toml
> [MPIPreferences]
> __clear__ = ["preloads_env_switch"]
> _format = "1.0"
> abi = "OpenMPI"
> binary = "system"
> cclibs = []
> libmpi = "libmpi"
> mpiexec = "mpiexec"
> preloads = []
> ```
> 
> either delete that file, or just disable the shared depot for now.

Yes, I just arrived at the same conclusion, which also explains why it was working when the depot was empty. 

I'll clean up this PR, clean the depot, and try again.

**Pull request merged**